File: Source/Aladdin.CAPI.Environment.Java/src/aladdin/capi/environment/ConfigFactory.java
package aladdin.capi.environment;
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Элемент описания фабрики классов
///////////////////////////////////////////////////////////////////////////
public class ConfigFactory implements Serializable
{
    private static final long serialVersionUID = 8873238961263357395L;
    
    // имя фабрики и класса фабрики
    private final String name; private final String className; 
    
    // конструктор
    public ConfigFactory(String name, String className)
    {
        // сохранить переданные параметры
        this.name = name; this.className = className; 
    }
    // имя элемента
    public final String name() { return name; }
    // класс фабрики
    public final String className() { return className; } 
}

File: Source/Aladdin.CAPI.Environment/Environment/ConfigFactory.cs
using System;
using System.IO;
using System.Xml;

namespace Aladdin.CAPI.Environment
{
    ///////////////////////////////////////////////////////////////////////////
    // Элемент описания фабрики классов
    ///////////////////////////////////////////////////////////////////////////
    [Serializable]
    public class ConfigFactory 
    {
        // имя фабрики и класса фабрики
        private string name; private string className; 
    
        // конструктор
        public ConfigFactory(XmlElement element)
        {
            // получить имя элемента 
            name = element.GetAttribute("name"); 
            // проверить наличие имени элемента
            if (name.Length == 0) throw new IOException(); 
        
            // получить класс фабрики
            className = element.GetAttribute("class"); 
            // проверить наличие класса фабрики
            if (className.Length == 0) throw new IOException(); 
        }
        // имя элемента
        public string Name { get { return name; }}
        // класс фабрики
        public string Class { get { return className; }}
    }
}